mithril-common/src/test_utils/mock_extensions.rs

```diff
@@ -0,0 +1,61 @@
+//! A set of tools for working with `automock`
+//!
+//! IMPORTANT: To avoid polluting production code, those tools do not expose or reexpose any
+//! `automock`types, users need to add them themselves to their crates.
+use std::sync::Arc;
+
+/// Helper to create configured Mockall mock.
+///
+/// This allows creation of the mock in a dedicated block isolated from the remaining test method
+/// code.
+pub struct MockBuilder<M: Default> {
+    phantom: std::marker::PhantomData<M>,
+}
+
+impl<M: Default> MockBuilder<M> {
+    /// Create a new instance of the mock with the given configuration
+    ///
+    /// The type must be specified either:
+    /// ```
+    /// use mithril_common::test_utils::mock_extensions::MockBuilder;
+    /// # #[derive(Default)] struct MockType {};
+    ///
+    /// // from the builder generic
+    /// let mock = MockBuilder::<MockType>::configure(|mock| {});
+    ///
+    /// // or from the closure parameter
+    /// let mock = MockBuilder::configure(|mock: &mut MockType| {});
+    /// ```
```

Comment:
It sounds like the way generic types and their inference work in Rust. Do we need to keep it?